### PR TITLE
day2

### PR DESCRIPTION
**Pull request by ShayRasooly**

### CODE REVIEW

File: Week1/Day2/DailyChallenge/code.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+print(f"The list of multiples of {number} with length {length} is:")
+print(multiples)
+
+#Challenge2
+user_word = input("Enter a word: ")
+
+new_word = ""
+
+for i in range(len(user_word)):
+    if i == len(user_word) - 1 or user_word[i] != user_word[i + 1]:
```
Comment (ShayRasooly) [inline Week1/Day2/DailyChallenge/code.py:18]: is there a way to shorten this condition? why bother checking each character if its the last one?

File: Week1/Day2/ExercisesXP/code.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
+#Exercise1
+my_fav_numbers = {7,88,24,23,10}
+my_fav_numbers.add(33)
+my_fav_numbers.add(16)
+print(my_fav_numbers)
+my_fav_numbers.remove(10) #we can delete last number only if we know what is last number
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:6]: we meant the last number you've added, you're right that set doesn't have order

File: Week1/Day2/ExercisesXP/code.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+#Exercise2
+sample_tuple = (1,2,3,4,5,6,7,8,9) #it is not possible to add more integers to a tuple
+
+#Exercise3
+basket = ["Banana", "Apples", "Oranges", "Blueberries"]
+basket.remove("Banana")
+basket.remove("Blueberries")
+basket.append("Kiwi")
+basket.insert(0, "Apples")
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:21]: any other way to add an element at the start of the list?

File: Week1/Day2/ExercisesXP/code.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    float_list.append(float)
+print(float_list)
+print("---------")
+#Exercise5
+for i in range(1,21):
+    print(i)
+print("---------")
+
+for i in range(1,21):
+    if i % 2 == 0:
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:40]: is the number the same as his index? what is the first index and what is the first number?

File: Week1/Day2/ExercisesXP/code.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+my_name = "Gregory"
+user_name = ""
+
+while user_name != my_name:
+    user_name = (input("What is your name? "))
+    user_name = user_name.capitalize()
+print("We are namesakes!")
+
+#Exercise7
+user_fav_fruits = input("What kind of fruit do you like? \nPlease separate the fruits with a single space\n")
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:54]: fruit or fruits? :)

File: Week1/Day2/ExercisesXP/code.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+list_of_toppings = []
+topping = ""
+while topping != "quit":
+    topping = (input("What topping do you want to add? "))
+    if topping != "quit":
+        list_of_toppings.append(topping)
+        print("Okay, I will add " + topping + " to your pizza")
+
+cost_of_pizza = 10 + (2.5*len(list_of_toppings))
+print(f"Your pizza will be with {', '.join(list_of_toppings)} and will cost ${cost_of_pizza}")
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:75]: nice usage of join

File: Week1/Day2/ExercisesXP/code.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+#Exercise9
+total_ticket_price = 0
+age = int
+while age != 0:
+    age = int(input("Enter the age of each person \nEnter 0 if you finish\n"))
+    if age != 0 and age > 0:
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:82]: why check twice if the age is zero? and what about babies whose age is zero?

File: Week1/Day2/ExercisesXP/code.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        if age < 3:
+            total_ticket_price += 0
+        elif age in range(3, 13):
+            total_ticket_price += 10
+        elif age > 12:
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:87]: Why Elif and not else?

File: Week1/Day2/ExercisesXP/code.py [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+            total_ticket_price += 15
+
+print(f"Your total price is ${total_ticket_price}")
+
+
+names = ["Alice", "Bob", "Charlie", "David", "Emma", "Frank"]
+
+allowed_names = []
+
+for name in names:
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:97]: You're deciding for people what is their name? you could've just asked

File: Week1/Day2/ExercisesXP/code.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+#Exercise 10
+sandwich_orders = ["Tuna sandwich", "Pastrami sandwich", "Avocado sandwich", "Pastrami sandwich", "Egg sandwich", "Chicken sandwich", "Pastrami sandwich"]
+
+while "Pastrami sandwich" in sandwich_orders:
+    sandwich_orders.remove("Pastrami sandwich")
+
+finished_sandwiches = []
+
+while sandwich_orders:
+    current_sandwich = sandwich_orders.pop(0)
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/code.py:115]: nice usage of pop

File: Week1/Day2/DailyChallenge/code.py (hunk 1 above)
Comment (ArkhipovG) [reply Week1/Day2/DailyChallenge/code.py:18]: maybe with var "prev_char"

File: Week1/Day2/ExercisesXP/code.py (hunk 3 above)
Comment (ArkhipovG) [reply Week1/Day2/ExercisesXP/code.py:21]: basket = ["Apples", *basket]

File: Week1/Day2/ExercisesXP/code.py (hunk 2 above)
Comment (ArkhipovG) [reply Week1/Day2/ExercisesXP/code.py:6]: my_fav_numbers.remove(16)

File: Week1/Day2/ExercisesXP/code.py (hunk 4 above)
Comment (ArkhipovG) [reply Week1/Day2/ExercisesXP/code.py:40]: no, the first index is 0, the first number is 1

File: Week1/Day2/ExercisesXP/code.py (hunk 5 above)
Comment (ArkhipovG) [reply Week1/Day2/ExercisesXP/code.py:54]: fruits)

File: Week1/Day2/ExercisesXP/code.py (hunk 7 above)
Comment (ArkhipovG) [reply Week1/Day2/ExercisesXP/code.py:82]: I should use -1 instead of 0

File: Week1/Day2/ExercisesXP/code.py (hunk 8 above)
Comment (ArkhipovG) [reply Week1/Day2/ExercisesXP/code.py:87]: last should be else without condition

File: Week1/Day2/ExercisesXP/code.py (hunk 9 above)
Comment (ArkhipovG) [reply Week1/Day2/ExercisesXP/code.py:97]: in the task, it's said that we have the list. "Given a list of names, write a program that asks teenager for their age, if they are not permitted to watch the movie, remove them from the list."

**Pull request merged**